flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -1046,10 +1046,16 @@ TimerService getTimerService() {
 		return timerService;
 	}
 
-	public ProcessingTimeService getProcessingTimeService(int operatorIndex) {
-		Preconditions.checkState(timerService != null, "The timer service has not been initialized.");
-		MailboxExecutor mailboxExecutor = mailboxProcessor.getMailboxExecutor(operatorIndex);
-		return new ProcessingTimeServiceImpl(timerService, callback -> deferCallbackToMailbox(mailboxExecutor, callback));
```

Comment:
Why was this removed? Yes, the method is ugly and the solution would be to properly use `StreamOperatorFactory` concept, but the deletion of this method is having a very nasty effect, where `ProcessingTimeServiceAware` can be applied to a factory AND to an operator, which creates a confusion by allowing for a bit pointless patterns, where operator has a factory and both the operator and it's factory are `ProcessingTimeServiceAware`.

Unless I'm missing something, I think it would be better for the API to keep this method, mark it `@Deprecated`, and never allow for `ProcessingTimeServiceAware` to be applied on an operator.

Reply:
If we keep this method, we need to add a map in `StreamTask` to store the corresponding `ProcessingTimeService` of all operators. In order to properly handle `quiesce` of `ProcessingTimeService`, the `ProcessingTimeService` obtained by passing the same `operatorIndex` must always be the same. The idea of introducing `ProcessingTimeServiceAware` is to store the `ProcessingTimeService` object in the operator. What do you think? @pnowojski